### PR TITLE
Plugins: Add state logs for plugin client retrieval

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**

- Simplifies proto client
- Adds logs for the plugin client fetching 
- Updates error messaging

**Why do we need this feature?**

In order to get better insight into what state the plugin is in when the client is requested.

### CODE REVIEW

File: pkg/plugins/backendplugin/grpcplugin/grpc_plugin.go [hunk 1]
```diff
@@ -134,94 +144,111 @@ func (p *grpcPlugin) Target() backendplugin.Target {
...
 }
 
-func (p *grpcPlugin) getPluginClient() (pluginClient, bool) {
+func (p *grpcPlugin) getPluginClient(ctx context.Context) (*ClientV2, bool) {
 	p.mutex.RLock()
-	if p.client == nil || p.client.Exited() || p.pluginClient == nil {
+	if p.client != nil && !p.client.Exited() && p.pluginClient != nil {
 		p.mutex.RUnlock()
-		return nil, false
+		return p.pluginClient, true
```
Comment (marefr) [inline pkg/plugins/backendplugin/grpcplugin/grpc_plugin.go:151]: Race problem here reading plugin client after read lock released. Put the client in a variable and then unlock as we did earlier

Comment (wbrowne) [reply pkg/plugins/backendplugin/grpcplugin/grpc_plugin.go:151]: Realized I can just `defer` here instead

**Review by andresmgot — APPROVED** [review]
LGTM

**Review by marefr — APPROVED** [review]
LGTM I think. but hard to know 😄 Expect you to have taken this for a spin and verified it works as expected

**Comment by wbrowne** [comment]
> LGTM I think. but hard to know 😄 Expect you to have taken this for a spin and verified it works as expected

Yep. All looks good. Will take it for a final spin to be doubly sure 🫡

**Pull request merged**